lib/_http_client.js

```diff
@@ -182,7 +182,7 @@ function ClientRequest(options, cb) {
                       'client');
     }
     self._storeHeader(self.method + ' ' + self.path + ' HTTP/1.1\r\n',
-                      self._headers);
+                      self._headersStore);
```

Comment:
> it might be worth sticking to properties for now

I don't think so. The numbers are very close and on multiple runs, the order varies a lot. It would be worth only for v4.x